config/routes.rb

```diff
@@ -9,4 +9,16 @@
   resources :users do
     resources :posts
   end
+
+  resources :posts
+
+  resources :posts do
+    resources :comments
+  end
+  post '/posts/new', to: 'posts#create' #create_new_post
+
+  resources :posts do 
+    resources :likes
+  end
+
```

Comment:
- [ ] A lot of your routes are repeated. Apart from being confusing, it is not the best practice. Please clean up your routes, especially the ones related to `posts`. Also remember to only expose methods that you would be using.
I will give you a hint 👇 
```ruby
resources :users, only: [:index, :show] do
  resources :posts, only: [:index, :show, :new, :create] do
    resources :comments, only: [:new, :create]
    resources :likes, only: [:create]
  end
end

```